ffmpeg_tools/validation.py

```diff
@@ -116,6 +131,7 @@ def validate_data_and_subtitle_streams(
 def validate_transcoding_params(
     dst_params,
     src_metadata,
+    dst_muxer_info,
```

Comment:
I would prefere that muxer_info, were optional. If someone passes None, we don't validate target audio codec.
We don't want to force someone to pass such exotic information.
For example in our transcoding webservice, we don't want to query it.

Reply:
The thing is that this information is not really optional. At least not if you want to validate sample rates properly. I see your point - this information should preferably by obtained by ffmpeg-tools on its own and not visible to the user. The problem is that the validation is not running inside the container and may not have access to ffmpeg. So making it optional only makes it possible for the user to step around validations that are there to detect early that his file won't be handled properly.

I don't know much about how the webservice is works though - does it run Golem code (with `ffmpeg-scripts`, Docker container and all) or does it just use ffmpeg-tools directly on host? If it's the latter, it would make sense to have an option for ffmpeg-tools to run `query_encoder_info()` on its own and get that info without bothering the user. But otherwise it's not safe.

What's your decision? I can change it to work the way you prefer but please be aware of the consequences.

Reply:
Quering muxer info optionally, would be good option, but we had to change api, to pass filename to ffmpeg.
I'm aware of consequences. Note that we can't pass through validation, because everything will go through Golem integration.
And if someone wants to violate validation API, he can. It's enought pass strip streams bools, to validation and to not pass them to replace step.

And I don't like this change either, but I really don't want to force webservice to query muxer info :(

Reply:
Here's a commit that makes it optional: 963b50c66f72526bfe20af190fe42710fe4cb76b. It disables the checks for target audio codec when it's not specified and we cannot guess it because `muxer_info` is unavailable.